### ISSUE TITLE
a bug on a sub task val accuracy when training on multi GPUs

### ISSUE THREAD

**Issue by wangshushu0213** [lyuqin/HydraNet-WikiSQL#4]
When I train the model on  multi GPUs, a sub task metric val accuracy is extremely low while the other metrics are high. (python main.py train --conf conf/wikisql.conf --gpu 0,1,2,3 --note "some note")
But when I train it on single GPU, the bug disappears.(python main.py train --conf conf/wikisql.conf --gpu 0 --note "some note")

 I have to use a small batch size to train on a single GPU so the performance is 3% lower in dev/test logical form acc. I want to train it on multi GPUs but I don't know why the bug occurs and how to solve it?

![训练111](https://user-images.githubusercontent.com/83213834/116068123-7744eb80-a6bc-11eb-8637-ec5b99ea2f65.png)
![预测111](https://user-images.githubusercontent.com/83213834/116067814-1b7a6280-a6bc-11eb-817e-404e2321731c.png)

**Comment by lyuqin** [comment]
This is a known issue when using newer transformers library. Can you double check transformers==2.3.0?

The reason is that in newer version of transformers, the following line won't get the correct column_token_length due to tokenize_result["token_type_ids"] is deprecated. 
https://github.com/lyuqin/HydraNet-WikiSQL/blob/566afd926d31e03ef09df6e8f647f8e69ecec3a2/featurizer.py#L139

**Comment by lyuqin** [comment]
But I am not sure why it doesn't work on multi-gpu while it **works** on single-gpu. I assume it should both succeed/fail. Moreover, the overall accuracy could fluctuate compare to the released model by ~0.3%, but there should be something wrong if it's 3% lower. 
I don't have such case from all the user feedbacks. Since I can't reproduce this in my environment, it's hard for me to find out the root cause.....

**Comment by lyuqin** [comment]
I updated the code and model. Please checkout the readme, and pull the latest master and download the model again. It has fixed the issue mentioned above

**Comment by wangshushu0213** [comment]
Thank you very much for your reply. I solved the multi-GPU problem using your last model version in the afternoon (Beijing time). I think it is the pytorch version and transformers version that caused my problem. My environment is : pytorch 1.7.1 transformers 2.3.0 and it errors when using multi-GPU, I searched on the net and revisised the modeling.py in transformers package and the problem was solved.
I notice that the model performs better when training with multi-GPU (I use batch size=64 when using 2 GPUs while I use 32 when training on single GPU).
